provider/azure/environ_test.go

```diff
@@ -777,20 +778,51 @@ func (s *environSuite) assertStartInstanceRequests(
 			Priority:                 to.Int32Ptr(100),
 			Direction:                network.SecurityRuleDirectionInbound,
 		},
-	}, {
-		Name: to.StringPtr("JujuAPIInbound"),
-		SecurityRulePropertiesFormat: &network.SecurityRulePropertiesFormat{
-			Description:              to.StringPtr("Allow API connections to controller machines"),
-			Protocol:                 network.SecurityRuleProtocolTCP,
-			SourceAddressPrefix:      to.StringPtr("*"),
-			SourcePortRange:          to.StringPtr("*"),
-			DestinationAddressPrefix: to.StringPtr("192.168.16.0/20"),
-			DestinationPortRange:     to.StringPtr("17777"),
-			Access:                   network.SecurityRuleAccessAllow,
-			Priority:                 to.Int32Ptr(101),
-			Direction:                network.SecurityRuleDirectionInbound,
-		},
 	}}
+	if args.autocert {
+		securityRules = append(securityRules, network.SecurityRule{
```

Comment:
A few comments saying what all the magic numbers are for might be good. e.g. that 17777 is the API port number configured elsewhere etc.

Reply:
Will do.

Reply:
I got rid of the magic number.